### ISSUE TITLE
Leiningen 2.9.0 breaks codox

### ISSUE THREAD

**Issue by gigasquid** [weavejester/codox#181]
To reproduce:

with lein 2.9.0 do

`lein new foo`
`cd foo`
add the plugin to the project.clj

```(defproject foo "0.1.0-SNAPSHOT"
  :description "FIXME: write description"
  :url "http://example.com/FIXME"
  :license {:name "EPL-2.0 OR GPL-2.0-or-later WITH Classpath-exception-2.0"
            :url "https://www.eclipse.org/legal/epl-2.0/"}
  :dependencies [[org.clojure/clojure "1.10.0"]]
    :plugins [[lein-codox "0.10.5"]
              [lein-cljfmt "0.5.7"]]
  :repl-options {:init-ns foo.core})
```

Run `lein codox`

```clojure.lang.Compiler$CompilerException: Syntax error macroexpanding clojure.core/fn at (clojure/core/unify.clj:83:18).
#:clojure.error{:phase :macro-syntax-check, :line 83, :column 18, :source "clojure/core/unify.clj", :symbol clojure.core/fn}
 at clojure.lang.Compiler.checkSpecs (Compiler.java:6971)
    clojure.lang.Compiler.macroexpand1 (Compiler.java:6987)
    clojure.lang.Compiler.analyzeSeq (Compiler.java:7092)
    clojure.lang.Compiler.analyze (Compiler.java:6789)
```
```Caused by: clojure.lang.ExceptionInfo: Call to clojure.core/fn did not conform to spec.
#:clojure.spec.alpha{:problems ({:path [:fn-tail :arity-1 :params], :pred clojure.core/vector?, :val clojure.core.unify/var-unify, :via [:clojure.core.specs.alpha/params+body :clojure.core.specs.alpha/param-list :clojure.core.specs.alpha/param-list], :in [0]} {:path [:fn-tail :arity-n], :pred (clojure.core/fn [%] (clojure.core/or (clojure.core/nil? %) (clojure.core/sequential? %))), :val clojure.core.unify/var-unify, :via [:clojure.core.specs.alpha/params+body :clojure.core.specs.alpha/params+body], :in [0]}), :spec #object[clojure.spec.alpha$regex_spec_impl$reify__2509 0x49dc7102 "clojure.spec.alpha$regex_spec_impl$reify__2509@49dc7102"], :value (clojure.core.unify/var-unify [G__794 G__795 G__796 G__797] (clojure.core/if-let [vb__787__auto__ (G__797 G__795)] (clojure.core.unify/garner-unifiers G__794 vb__787__auto__ G__796 G__797) (clojure.core/if-let [vexpr__788__auto__ (clojure.core/and (G__794 G__796) (G__797 G__796))] (clojure.core.unify/garner-unifiers G__794 G__795 vexpr__788__auto__ G__797) (if (clojure.core.unify/occurs? G__794 G__795 G__796 G__797) (throw (java.lang.IllegalStateException. (clojure.core/str "Cycle found in the path " G__796))) (clojure.core.unify/bind-phase G__797 G__795 G__796))))), :args (clojure.core.unify/var-unify [G__794 G__795 G__796 G__797] (clojure.core/if-let [vb__787__auto__ (G__797 G__795)] (clojure.core.unify/garner-unifiers G__794 vb__787__auto__ G__796 G__797) (clojure.core/if-let [vexpr__788__auto__ (clojure.core/and (G__794 G__796) (G__797 G__796))] (clojure.core.unify/garner-unifiers G__794 G__795 vexpr__788__auto__ G__797) (if (clojure.core.unify/occurs? G__794 G__795 G__796 G__797) (throw (java.lang.IllegalStateException. (clojure.core/str "Cycle found in the path " G__796))) (clojure.core.unify/bind-phase G__797 G__795 G__796)))))}
 at clojure.spec.alpha$macroexpand_check.invokeStatic (alpha.clj:705)
    clojure.spec.alpha$macroexpand_check.invoke (alpha.clj:697)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.AFn.applyTo (AFn.java:144)
```

**Comment by weavejester** [comment]
Should be fixed in 0.10.6.

**Comment by gigasquid** [comment]
Awesome! thanks for fixing so fast!
